sabnzbd/nzbstuff.py

```diff
@@ -1187,14 +1190,22 @@ def remove_article(self, article: Article, success: bool):
 
         # Check if we can succeed when we have missing articles
         # Skip check if retry or first articles already deemed it hopeless
-        if not success and job_can_succeed and not self.reuse and cfg.fail_hopeless_jobs():
-            job_can_succeed, _ = self.check_availability_ratio()
+        if not success and job_can_succeed and not self.reuse:
+            # Abort if more than 50% is missing after reaching missing_threshold_mbytes
+            job_can_succeed = self.check_missing_threshold_mbytes()
```

Comment:
This should only be called if `cfg.fail_hopeless_jobs() == True`, like the other checks.